File: src/cmcandy/java_language_answers/_0242.java
package cmcandy.java_language_answers;

import java.util.Arrays;

/**
 * Created by CHEN on 2020/5/15.
 */
public class _0242 {
    static class Solution {
        public boolean isAnagram(String s, String t) {
            if(s.equals(t))return true;
            if (t.length()!=s.length()) return false;
            char[] st = s.toCharArray();
            char[] tt = t.toCharArray();
            Arrays.sort(st);
            Arrays.sort(tt);
            return String.valueOf(st).equals(String.valueOf(tt));
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isAnagram("a","b") );
    }
}

File: src/cmcandy/C_Language_Answers/_0242.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace ::std;


class Solution {
public:
    bool isAnagram(string s, string t) {
            int len = s.length();
            if (t.length()!=len)
            {
                return false;
            }
            vector<int> v(26,0);

            for (int i = 0; i < len; i++)
            {
               v[s[i]-'a']++;
               v[t[i]-'a']--;
            }
            for (int i = 0; i < 26; i++)
            {
                if(v[i]!=0) return false;
            }
            
            return true;
    }
};